### DISCUSSION TITLE
Add ability for iOS users to switch app icon

### DISCUSSION THREAD

**Discussion by JanOstrowka** [ente-io/ente#843] [Enhancements]
## Summary

It would be great if iOS users could choose between different app icons for the Gallery app. Feature can be found in other iOS apps like Apollo or Home Assistant.

I've included a few alternative icon propositions below:
(I'm happy to mock additional ones)

- Original iOS Gallery icon featuring sunflower
![oldschool](https://github.com/ente-io/photos-app/assets/3185243/01b4c98a-5a81-4862-83e7-330f1de5661e)

- Recreated version of the current Gallery icon
![image](https://github.com/ente-io/photos-app/assets/3185243/ba220cf0-f8b6-44b3-be23-6c25dffe3dd7)

- Gallery icon with ente flavor
![image](https://github.com/ente-io/photos-app/assets/3185243/2b8c75b3-2242-410d-b243-4d2311cfe528)

- "e" shaped flower (my personal favourite)
![image](https://github.com/ente-io/photos-app/assets/3185243/7e89e9e2-a125-43f2-a839-ff6606dc67b0)

## Expected Behavior

- A new "App Icon" section in Settings on iOS 
- Options to select between the 5 icons (original and 4 alternative, mentioned above)
- Selected icon dynamically changes without needing to restart the app

## Benefits

- More customisation options for iOS users
- Ability to make the icon look like the default gallery

**Comment by vishnukvmd** [comment]
Hey @JanOstrowka, thanks for the suggestion (and the icons!)

There's a clear path forward for iOS with [flutter_dynamic_icon](https://pub.dev/packages/flutter_dynamic_icon), Android seems a bit iffy but could explore [android_dynamic_icon](https://pub.dev/packages/android_dynamic_icon).
